pkg/scheduler/worker.go

```diff
@@ -121,10 +112,29 @@ func (w *Worker) process(component *keb.Components, state cluster.State, schedul
 			return false, err
 		}
 		return false, nil
-	case StateNew, StateInProgress, StateFailed:
+	case StateNew, StateInProgress:
+		// Operation still being processed by the component reconciler
```

Comment:
Currently we change the status in two different places... failures are handled in the workers and successful-runs are handled in the scheduler.

Could be worth to move the status-update logic into one place. One option could be to implement a small and simple state-machine which will be updated by the workers (e.g. using a channel) and the state-machine decides which cluster-state should be persisted in the database.